lib/cozy-light.js

```diff
@@ -1,1296 +1,93 @@
-        if (err) {
-          LOGGER.error('An error occured while stopping plugins');
-          LOGGER.raw(err);
-        }
-
-        LOGGER.info('Stopping server...');
-        mainAppHelper.stop(function(){
-          port = defaultAppsPort;
-          LOGGER.info('\t' + symbols.ok + '\tmain server');
-          if (callback) { callback(); }
-        });
-      });
     });
   },
-
-  /**
-   * Stops application server, its apps, and its plugins,
-   * reload everyone, then restart everyone
-   */
-  restart: function (callback) {
-    LOGGER.info('Restarting...');
-    actions.stop(function(){
-      configHelpers.loadConfigFile();
-      loadedPlugins = {};
-      loadedApps = {};
-      routes = {};
-      LOGGER.info('Starting all apps...');
-      actions.start(program, function(){
-        LOGGER.info('...Cozy Light was properly restarted.');
-        if (callback) { callback(); }
-      });
-    });
-  },
-
   /**
    * Manage properly exit of the process when SIGINT signal is triggered.
    * It asks to every plugin to end properly.
    */
-  exit: function (callback) {
-    actions.stop(function endProcess (err) {
+  stop: function( callback ){
+    cozyLight.actions.stop(function endProcess (err) {
       if (err) {
-        LOGGER.error('Cozy Light was not properly terminated.');
+        cozyLight.logger.error('Cozy Light was not properly terminated.');
       } else {
-        LOGGER.info('Cozy Light was properly terminated.');
+        cozyLight.logger.info('Cozy Light was properly terminated.');
       }
 
-      if (db) {
+      var doneCallback = function(){
+        cozyLight.configHelpers.watcher.release();
+        cozyLight.nodeHelpers.invoke(callback);
+      };
+
+      if (cozyLight.db) {
         try {
-          if(!db._closed) {
-            db.close();
+          /*eslint-disable */
+          if(!cozyLight.db._closed) {
+            /*eslint-enable */
+            return cozyLight.db.close(doneCallback);
           }
         } catch (err) {
-          LOGGERR.error(err);
+          cozyLight.logger.error(err);
         }
       }
-
-      if (callback !== undefined) {
-        callback();
-      }
+      return cozyLight.nodeHelpers.invoke(doneCallback);
     });
   },
-
   /**
    * Manage properly exit of the process when SIGINT signal is triggered.
    * It asks to every plugin to end properly.
    */
-  exitNow: function () {
-    actions.exit(function killProcess (err) {
-      LOGGER.info('Killing process.');
-
-      /*eslint-disable */
-      var handlesC = process._getActiveHandles().length;
-      var requestsC = process._getActiveRequests().length;
-      /*eslint-enable */
-      if (process.env.DEBUG !== undefined && (handlesC + requestsC > 0)){
-        LOGGER.info('Has remaining open handles...');
-        LOGGER.info('_getActiveHandles ' + handlesC);
-        LOGGER.info('_getActiveRequests ' + requestsC);
-      }
+  exit: function( err ){
+    if ( !cozyLight.nodeHelpers.hasFreedResources() ){
```

Comment:
If we can make it a rule in eslint, that good to go, otherwise it putting effort in something difficult to manage too dependants of the people.